app/handlers/api.py

```diff
@@ -45,6 +45,108 @@
 from os.path import splitext
 
 
+class AutoCropperHandler(BaseHandler):
+    SUPPORTED_METHODS = ("GET", "POST")
+
+    @asynchronous
+    @engine
+    @web_authenticated
+    def get(self,):
+        response = yield Task(
+            self.api_call,
+            url=self.settings['API_URL'] + '/autocropper',
+            method='GET')
+        self.set_json_output()
+        if response.code == 404:
+            body = loads(response.body.decode("utf-8"))
+            self.finish(self.json_encode({
+                        'status': 'success',
+                        'message': body['message'],
+                        'data': []}))
+        else:
+            self.set_status(response.code)
+            self.finish(response.body)
+
+    @asynchronous
+    @engine
+    @web_authenticated
+    def post(self):
+
+
+        if self.request.files:
+            fileinfo = self.request.files['file'][0]
+            body = fileinfo['body']
+            fname = fileinfo['filename']
+            fileencoded = base64.b64encode(body).decode('utf-8')
+
+            body = {
+                "filename": fname,
+                "image": fileencoded,
+                'content_type': fileinfo['content_type']
+            }
+
+            response = yield Task(
+                self.api_call,
+                url=self.settings['API_URL'] + '/autocropper',
+                method='POST',
+                body=self.json_encode(body))
+            self.set_json_output()
+            self.set_status(response.code)
+            self.finish(response.body)
+
+
+class AutoCropperUploaderHandler(BaseHandler):
+    SUPPORTED_METHODS = ("POST",)
+
+
+    @asynchronous
+    @engine
+    @web_authenticated
+    def post(self):
+
+
+        if self.request.files:
+            fileinfo = self.request.files['file'][0]
+            body = fileinfo['body']
+            fname = fileinfo['filename']
+            info(fname)
+            dirfs = dirname(realpath(__file__))
+            info(dirfs)
+            fh = open(dirfs + '/' + fname, 'wb')
+            fh.write(body)
+            fh.close()
+            exif_data = get_exif_data(dirfs + '/' + fname)
+            with open(dirfs + '/' + fname, "rb") as imageFile:
+                fileencoded = b64encode(imageFile.read())
+            if fileencoded:
+                remove(dirfs + '/' + fname)
+                manual_coords = self.get_argument('manual_coords')
+                image_set_id = self.get_argument("image_set_id")
+
+                body = {
+                    "image_set_id": int(image_set_id),
+                    'manual_coords': manual_coords,
+                    "filename": fname,
+                    "exif_data": exif_data,
```

Comment:
what's exif_data?

Reply:
get_exif_data() this function extract the information about the image its already being use in existing flow